### ISSUE TITLE
more tree types and shrubs

### ISSUE THREAD

**Issue by bobombolo** [bobombolo/realterrain#43]
a few more tree types, perhaps one for each biome, a few more land cover node types custom to this mod, and maybe some shrubs, possibly even including some kelp for the water biome...

this could also be where multiple trees and shrubs per biome is implemented. there is room in the form...

**Comment by bobombolo** [comment]
the tail of #37 should move here

![screenshot_20151111_195733](https://cloud.githubusercontent.com/assets/12679496/11110265/798d0cba-88ae-11e5-9fff-98695de83cc5.png)

**Comment by bobombolo** [comment]
played with adding a cottage as a tree type for the urban biome. collissions like these are ok with trees but not so much with structures drawn randomly by biome
![screenshot_20151220_115123](https://cloud.githubusercontent.com/assets/12679496/11919768/5b676dae-a710-11e5-80c0-21cd8d8e2a0b.png)

also the shrubs don't detect for anything other than a single block trunk so this happens:

![screenshot_20151220_115252](https://cloud.githubusercontent.com/assets/12679496/11919770/69297900-a710-11e5-9aba-50413367d045.png)

**Comment by bobombolo** [comment]
recently discovered the "register_schem" and "register_decoration" methods in the lua api.txt which might be helpful

**Comment by Gael-de-Sailly** [comment]
`minetest.register_decoration` may be difficult to use here, because it would place the decorations in all biomes. I think we should rather change the generation order.

**Comment by bobombolo** [comment]
yes there needs to be a collision detection system for the trees system, and for trees that have a footprint bigger than 1x1 there needs to be collision detection for the shrubs as well.

**Comment by bobombolo** [comment]
there is now a method for placing schems on voxelmanip: https://github.com/minetest/minetest/blob/62d15ac7c1fcd7214a9e45d46bbc560f998edb95/doc/lua_api.txt#L2346
